scripts/production/backup_master.pl

```diff
@@ -0,0 +1,143 @@
+taking its location from a production_reg_conf file.
+
+=head1 SYNOPSIS
+
+  perl backup_master.pl --reg_conf $COMPARA_REG_PATH
+
+  perl backup_master.pl $COMPARA_REG compara_master --label "pre100"
+
+=head1 ARGUMENTS
+
+=head2 DATABASE SETUP
+
+=over
+
+=item B<[--url mysql://user[:passwd]@host[:port]/dbname]>
+
+URL of the database to test.
+
+=item B<[--reg_conf registry_configuration_file]>
+
+The Bio::EnsEMBL::Registry configuration file. If none given and no URL is
+given, the one set in ENSEMBL_REGISTRY will be used if defined, if not
+~/.ensembl_init will be used.
+
+=item B<[--reg_type reg_type]>
+
+The "type" or "group" under which the database is to be found in the Registry.
+Defaults to "compara".
+
+=item B<[--reg_alias|--reg_name name]>
+
+The name or "species" under which the database is to be found in the Registry.
+Defaults to "compara_master".
+
+=back
+
+=head2 DUMP SETUP
+
+=over
+
+=item B<[--dump_path /path/to_dumps]>
+
+Where to store the dumps. Defaults to the shared warehouse directory.
+
+=item B<[--username username]>
+
+Name of the user used to create the dumps.
+Defaults to "compara_ensembl". Set this to an empty string to create the
+backup as yourself.
+
+=item B<[--label str]>
+
+Label to add to the file name
+
+=back
+
+=cut
+
+use strict;
+use warnings;
+
+use Getopt::Long;
+use POSIX qw(strftime);
+
+use Bio::EnsEMBL::Registry;
+
+use Bio::EnsEMBL::Compara::DBSQL::DBAdaptor;
+use Bio::EnsEMBL::Compara::Utils::Registry;
+
+my ($url, $reg_conf, $reg_type, $reg_alias);
+my ($dump_path, $username, $label);
+
+# Arguments parsing
+GetOptions(
+    'url=s'                         => \$url,
+    'reg_conf|regfile|reg_file=s'   => \$reg_conf,
+    'reg_type=s'                    => \$reg_type,
+    'reg_alias|regname|reg_name=s'  => \$reg_alias,
+    'dump_path=s'                   => \$dump_path,
+    'username=s'                    => \$username,
+    'label=s'                       => \$label,
+) or die "Error in command line arguments\n";
+
+if (@ARGV) {
+    die "ERROR: There are invalid arguments on the command-line: ". join(" ", @ARGV). "\n";
+}
+
+unless ($url or $reg_conf) {
+    print "\nNeither --url nor --reg_conf are defined. Some of those are needed to refer to the database being tested\n\n";
+    exit 1;
+}
+
+if ($reg_conf) {
+    Bio::EnsEMBL::Registry->load_all($reg_conf);
+}
+my $dba = $url
+    ? Bio::EnsEMBL::Compara::DBSQL::DBAdaptor->new( -URL => $url )
+    : Bio::EnsEMBL::Registry->get_DBAdaptor( $reg_alias || 'compara_master', $reg_type || 'compara' );
+
+my $division = $dba->get_division;
+
+my @params = (
+    '--host'    => $dba->dbc->host,
+    '--port'    => $dba->dbc->port,
+    '--user'    => 'ensro',
+);
+
+$dump_path //= '/nfs/production/panda/ensembl/warehouse/compara/master_db_dumps';
+$username  //= 'compara_ensembl';
+my $date = strftime '%Y%m%d', localtime;
+my $dump_name = "ensembl_compara_master_${division}.${date}" . ($label ? ".$label" : ''). ".sql";
```

Comment:
```suggestion
my $dump_name = "ensembl_${$reg_alias}_${division}.${date}" . ($label ? ".$label" : '') . ".sql";
```

Reply:
I am not expecting this script to be used for master databases owned by individuals, and in this case both `--username` and `--dump_path` would have to be given on the command line because the default values wouldn't be sensible.